sw/airborne/modules/computer_vision/video_usb_logger.c

```diff
@@ -35,34 +35,30 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
-
+#include "computer_vision/lib/encoding/jpeg.h"
+#include "pose_history/pose_history.h"
 
 /** Set the default File logger path to the USB drive */
 #ifndef VIDEO_USB_LOGGER_PATH
 #define VIDEO_USB_LOGGER_PATH /data/video/usb
 #endif
 
-#ifndef VIDEO_USB_LOGGER_WIDTH
-#define VIDEO_USB_LOGGER_WIDTH 272
-#endif
-
-#ifndef VIDEO_USB_LOGGER_HEIGHT
-#define VIDEO_USB_LOGGER_HEIGHT 272
-#endif
-
 /** The file pointer */
 static FILE *video_usb_logger = NULL;
-struct image_t img_jpeg;
+struct image_t img_jpeg_global;
+bool created_jpeg = FALSE;
 char foldername[512];
 int shotNumber = 0;
 
-void save_shot(struct image_t *img, struct image_t *img_jpeg)
+void save_shot_on_disk(struct image_t *img, struct image_t *img_jpeg);
+void save_shot_on_disk(struct image_t *img, struct image_t *img_jpeg)
```

Comment:
If it's only an internal function, just make it static... then you also don't need the declaration before it...